File: src/main/java/org/openmedstack/readmodels/IUpdateReadModel.java
package org.openmedstack.readmodels;

import org.openmedstack.MessageHeaders;
import org.openmedstack.events.BaseEvent;

import java.util.concurrent.CompletableFuture;

public interface IUpdateReadModel<T extends BaseEvent> {
    CompletableFuture update(T domainEvent, MessageHeaders headers);
}

File: src/main/kotlin/org/openmedstack/readmodels/IUpdateReadModel.kt
package org.openmedstack.readmodels

import org.openmedstack.MessageHeaders
import org.openmedstack.events.BaseEvent
import java.util.concurrent.CompletableFuture

interface IUpdateReadModel {
    fun canUpdate(eventType: Class<*>): Boolean
    fun update(domainEvent: BaseEvent, headers: MessageHeaders?): CompletableFuture<*>
}